### DISCUSSION TITLE
Cancel a long running function

### DISCUSSION THREAD

**Discussion by pyinstallerbaggins** [marimo-team/marimo#2245] [Q&A] [answered]
Anyone know of a way to cancel a running cell? For example, i triggered a query that executes a long running query but i want to cancel it without having to shut down the notebook.

mo.stop()?

**Comment by akshayka** [comment]
You can use the stop button

![image](https://github.com/user-attachments/assets/28da0287-b65a-42ba-96e0-f89ac83b8c71)

Also on the bottom right of the page

![image](https://github.com/user-attachments/assets/6c3052fb-be43-4847-b423-9a2e1db27d19)

**Reply by pyinstallerbaggins** [reply]
How would you stop it when deployed via marimo run

**Reply by akshayka** [reply, marked as answer]
Ah, that's not possible; fundamental limitation of our architecture. You might try to run queries remotely so at least they don't lock up your notebook.
